### ISSUE TITLE
Add MessagePack archiver

### ISSUE THREAD

**Issue by rggjan** [USCiLab/cereal#128]
A MessagePack (http://msgpack.org/) archiver would be great. This would have the advantage of being as cross-platform as the json archiver, however with much smaller memory requirements...

**Comment by ecorm** [comment]
> Sounds good. Yes, I know the two implementations. However, they are both not header only.

@rggjan, msgpack-c is indeed header-only if you use the C++ API

**Comment by rggjan** [comment]
@ecorm, yes I know, that's great! At the time of writing this was not the case yet, but msgpack-c added header only support to the newest version.

**Comment by Type1J** [comment]
Any progress here?

**Comment by AzothAmmo** [comment]
The pull request is at #138, we haven't made any progress in merging it yet.

**Comment by Type1J** [comment]
Having a cross language binary format (any of them) is very desirable for me. I'm going to try to work from the pull request fork until it's merged.

**Comment by kklouzal** [comment]
How does this perform, memory wise, compared to the included Binary and Portable Binary archives? Does/would MessagePack output smaller archives?

**Comment by stephanlachnit** [comment]
How about https://github.com/mikeloomisgg/cppack? It is a single header so much simpler than msgpack-c. The interface is also quite similar to cereal so _maybe_ it would be easy to do. One disadvantage is that the last commit was from 2019.

**Comment by Type1J** [comment]
@stephanlachnit When it works, it works, so that may be a good option for some people. I've been using Rust with serde, lately, and I've translated most of the C and C++ that I modify frequently to Rust, so it worked out well. The backends for serde give a variety of formats, and it doesn't require any more than an annotation at the top of a struct to make it work.
